frontend/src/lib/components/IntervalFilter/intervalFilterLogic.ts

```diff
@@ -33,6 +43,68 @@ export const intervalFilterLogic = kea<intervalFilterLogicType>({
         setFilters: ({ filters }, _, __, previousState) => {
             const { date_from, date_to } = filters
             const previousFilters = selectors.filters(previousState)
+
+            let activeUsersMath: BaseMathType.WeeklyActiveUsers | BaseMathType.MonthlyActiveUsers | null = null
+
+            // We disallow grouping by certain intervals for weekly active users and monthly active users views
+            // e.g. WAUs grouped by month. Here, look for the first event/action running WAUs/MAUs math and
+            // pass that down to the interval filter to determine what groupings are allowed.
+            for (const series of [...(values.filters.events || []), ...(values.filters.actions || [])]) {
+                if (series.math === BaseMathType.WeeklyActiveUsers) {
+                    activeUsersMath = BaseMathType.WeeklyActiveUsers
+                    break
+                }
+
+                if (series.math === BaseMathType.MonthlyActiveUsers) {
+                    activeUsersMath = BaseMathType.MonthlyActiveUsers
+                    break
+                }
+            }
+
+            const enabledIntervals: Intervals = { ...intervals }
+
+            if (activeUsersMath) {
+                // Disallow grouping by hour for WAUs/MAUs as it's an expensive query that produces a view that's not useful for users
+                enabledIntervals.hour = {
+                    label: 'Hour',
+                    newDateFrom: 'dStart',
+                    disabledReason:
+                        'Grouping by hour is not supported on insights with weekly and monthly active users.',
+                }
+
+                // Disallow grouping by month for WAUs as the resulting view is misleading to users
+                if (activeUsersMath === BaseMathType.WeeklyActiveUsers) {
+                    enabledIntervals.month = {
+                        label: 'Month',
+                        newDateFrom: '-90d',
+                        disabledReason: 'Grouping by month is not supported on insights with weekly active users.',
+                    }
+                }
+            }
+
+            actions.setEnabledIntervals(enabledIntervals)
+
+            // If the user just flipped an event action to use WAUs/MAUs math and their
+            // current interval is unsupported by the math type, switch their interval
+            // to an appropriate allowed interval and inform them of the change via a toast
+            if (values.interval && !!enabledIntervals[values.interval].disabledReason) {
+                const humanReadableMathType =
+                    activeUsersMath === BaseMathType.MonthlyActiveUsers ? 'Monthly active users' : 'Weekly active users'
+
+                if (values.interval === 'hour') {
+                    lemonToast.info(
+                        `${humanReadableMathType} does not support grouping by hour. Grouping by day instead.`
```

Comment:
In a toast message, let's state what just happened first, and _then_  explain:
<img width="428" alt="Screenshot 2023-01-18 at 13 40 00" src="https://user-images.githubusercontent.com/4550621/213174351-1e9015fe-bebb-48de-a5a2-2b949cd0a48d.png">